### ISSUE TITLE
Please fix Spek versioning, compatibility and publishing.

### ISSUE THREAD

**Issue by artem-zinnatullin** [spekframework/spek#170]
Currently Spek has veery crazy versioning principles. 

* We're currently on `1.0.25` which is known as pre-v1…
* `1.0.89+` known as post-v1, somewhere near to this version `DescribeBody` was renamed to `Dsl`, **compatibility broken**.
* `1.1.0-beta3` is version shown by badge in GitHub README, in this version `Dsl` was renamed to `SpecBody`, **compatibility broken again**.
* Spek website's [root page](http://spekframework.org) says that current version is `1.0`, but there are **no jars** with such version.
* Spek website's [documentation page](http://spekframework.org/docs/latest/) points to *latest* version which is `1.1.0-beta3` while Spek website's [downloads page](http://spekframework.org/downloads) points to `1.0` which is actually `1.0.89`.
* Also, there is `1.1.19` in both repos, but `1.1.0-beta3` pointed everywhere else as latest…
* And there is a [GitHub releases page](https://github.com/JetBrains/spek/releases) which only contains `1.1.0-beta3` and `v1`, no mentions of stable `1.0.89` or actually latest `v1.1.19`.

To get these versions you need to find [bintray repo](http://dl.bintray.com/jetbrains/spek/org/jetbrains/spek/spek-api/) which is used in `build.gradle` in docs example, some artifacts are published to [jcenter](https://jcenter.bintray.com/org/jetbrains/spek/spek-api/) and 1.0.25 and some other versions live in internal [jetbrains repo](http://repository.jetbrains.com/all/org/jetbrains/spek/spek/).

I mean thank you for Spek, we have lots of specs in the project, but we're trying to migrate to *latest* Spek and this is just insane…

**Comment by raniejade** [comment]
The most recent release is `1.1.0-beta3` which is a pre-release for `1.1`.  Previous versioning was `<major>.<minor>.<build-no>`, `build-no` is controlled by TC so every build increments it.  So starting from `1.1` we will be using semantic versioning, `<major>.<minor>.<patch>[-<qualifier>]` and for pre-release we add a qualifier. Bintray will contain pre and major releases, but only major releases gets sync to central.

We just started using github's releases page to track changes starting from `1.1.0-beta3` that's why you don't see previous version there. Moving forward, we will place the change notes here and have a link in the homepage and/or docs pointing to it. Speaking of the homepage and docs, we are currently revamping them https://spekframework.github.io/).

**Comment by artem-zinnatullin** [comment]
Ok, we can help you with some of these things since we depend on Spek a lot, but thing that worries some of us about Spek is that development process happens mostly without Pull Requests and issues which complicates following changes, providing feedback on code reviews and staying up to date with the project for outside contributors.

**Comment by raniejade** [comment]
+1 on the pull requests, its a better place for feedback regarding the changes. Really appreciate the help, especially with new features and major changes.

**Comment by artem-zinnatullin** [comment]
That's very great to hear! We should be able to review PRs within 24h windows every work day.

Let's keep this issue open until all major points fixed? (consistent version in docs and GitHub releases page, description which maven repo should be used and when).

**Comment by raniejade** [comment]
Sure, we will keep this open.

**Comment by raniejade** [comment]
Btw, just to be clear. While we do appreciate feedbacks, ultimately the core members have the final say. Our decisions will be based on those feedbacks and the vision we have for Spek.

**Comment by artem-zinnatullin** [comment]
Sure!

**Comment by msogrin** [comment]
> Don't use 1.1.19 it's a remnant of the previous versioning and we will be deleting it from bintray.

It was said almost a year ago. Is there any chance the version 1.1.19 gets deleted?

**Comment by magneticflux-** [comment]
Perhaps [Shipkit](http://shipkit.org/) would be useful for you for versioning? It recently had it's first stable release, but Mockito has been using it for years.

**Comment by patrikerdes** [comment]
Running gradle-versions-plugin on my project I get
```
The following dependencies have later release versions:
 - org.jetbrains.spek:spek-api [1.1.5 -> 1.1.19]
 - org.jetbrains.spek:spek-junit-platform-engine [1.1.5 -> 1.1.19]
```
Please delete 1.1.19.

**Comment by PHPirates** [comment]
Yeah is this **very** confusing. I have `https://dl.bintray.com/jetbrains/spek` repo in my build.gradle in my build.gradle from the past as otherwise Spek can't be found, as in #139, so I looked there for latest version (just by hand, going to the link) which seemed to be 1.1.19, also as found by Patrik's plugin as he said. I checked also at `https://jcenter.bintray.com/org/jetbrains/spek/spek-api/` also contains both 1.1.5 and 1.1.19, although just using that as source won't find Spek 1.1.19 for me though it seems to be there. The good news is, that does seem to work again to find 1.1.5.

For reference, when I used 1.1.19 I got
```
java.lang.NoClassDefFoundError: org/junit/platform/engine/discovery/ClasspathSelector
Caused by: java.lang.ClassNotFoundException: org.junit.platform.engine.discovery.ClasspathSelector
```
